README.md

```diff
@@ -1,7 +1,153 @@
 # Go stats and tracing collection libraries
 
-[![Build
-Status](https://travis-ci.org/census-instrumentation/instrumentation-go.svg?branch=master)](https://travis-ci.org/census-instrumentation/instrumentation-go)
+# Go stats and tracing collection libraries
+
+[![Build Status][travis-image]][travis-url] [![GoDoc][godoc-image]][godoc-url]
 
 This is still at a very early stage of development and a lot of the API calls
 are in the process of being changed and will certainly break your code.
+
+[travis-image]: https://travis-ci.org/census-instrumentation/opencensus-go.svg?branch=master
+[travis-url]: https://travis-ci.org/census-instrumentation/opencensus-go
+[godoc-image]: https://godoc.org/github.com/census-instrumentation/opencensus-go?status.svg
+[godoc-url]: https://godoc.org/github.com/census-instrumentation/opencensus-go
+
+# Go stats core library
+
+## Keys and tagging
+
+### To create/retrieve a key
+A key is defined by its name. To use a key a user needs to know its name and type (currently only keys of type string are supported. Later support for keys of type int64 and bool will be supported).
+To create/retrieve a key the user calls a keys manager appropriate function. Keys are safe to be reused and called from multiple go routines.
+
+Create/retrieve key:
+
+    key1 := tagging.DefaultKeyManager().CreateKeyString("keyNameID1")
+    key2 := tagging.DefaultKeyManager().CreateKeyInt64("keyNameID2")
+    ...
+
+### Create a set of tags associated with keys
+To create a new tag set from scratch using changes:
+
+    change1 := key1.CreateChange("tagValue", TagOpUpsert)   // key1 is a KeyString so its value is a string
+    change2 := key2.CreateChange(10, TagOpInsert)           // key2 is a KeyInt64 so its value is an int64
+    builder := stats.NewTagsBuilder()
+    builder.StartFromEmpty()
+    builder.Apply(c1,c2)
+    tags := builder.Build()
+
+To create a new tag set from scratch using direct values (fastest):
+
+    builder := stats.NewTagsBuilder()
+    builder.StartFromEmpty()
```

Comment:
Do we really need it? Aren't we starting empty all the time?

Reply:
No. Most of the time you need to start from an existing tagSet that you want to add to or update. We had to kill this branch so PTAL @ the newly active pull request.
The readme.md file wasn't up to date. I updated it in the new pull request.